### PR TITLE
Milestone4 - Filter books

### PR DESCRIPTION
**Pull request by janis-jenny**
- Create the CHANGE_FILTER action in actions/index.js
- Create a filter reducer that modifies the filter in the Redux store.
- Implement the CategoryFilter component
- Keep the filter value in the Redux store
- Render the CategoryFilter component in the BooksList component.
- Implement the handleFilterChange method which modifies the filter state in the Redux store. Pass it down as a prop to CategoryFilter.
- Implement a conditional method that decides which books should be shown in the table depending on the filter value from Redux store.

### CODE REVIEW

File: src/reducers/filter.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    case CHANGE_FILTER:
+      return {
+        filter: action.payload,
+      };
+    default:
+      return state;
+  }
+};
+
+export default filterReducer; */
```
Comment (eliasecasta) [inline src/reducers/filter.js]: - [ ] One of the requirements that this project has is to default your filter.js to default which I can see that you did here but all the code is commented, what happened in this part? Did you add it elsewhere? This script is a project requirement and should be used. It defaults to 'All' [screenshot from Pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/ItDefaultsToAll.png)

File: src/reducers/index.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/* import { combineReducers } from 'redux';
+import reducerBooks from './books';
+import filterReducer from './filter';
+
+const rootReducer = combineReducers({
+  books: reducerBooks,
+  filter: filterReducer,
+});
+
+export default rootReducer; */
```
Comment (eliasecasta) [inline src/reducers/index.js]: - [ ] Please remove all the commented code that you may have to keep the codebase clean.

**Review by eliasecasta — CHANGES_REQUESTED** [review]
# Status:  Changes Required ♻️
_Check the comments under the review_
<hr>

Hello @janis-jenny, I am Elias, your reviewer for this PR

Almost everything is working as expected, great job so far with your project!

I will mention to you a small summary of things that need to be changed in order to keep moving:

- [ ] Right now, when you boot your project or refresh the page, this happens:
image 

- In order to fix this, I need to switch the category filter to something else than "All" and then come back to "All" so I can see your book samples, please fix this:
image

Everything else looks awesome!

Please correct those issues and send another review request, good luck, and well done so far! 

Code review by Elias Castañeda

_(Feel free to leave any questions or comments in the PR thread if something is not 100% clear)_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/containers/Booklist.js [hunk 3]
```diff
@@ -1,32 +1,53 @@
-import React from 'react';
-import { useStore } from 'react-redux';
+/* eslint-disable no-console */
```
Comment (rukundoeric) [inline src/containers/Booklist.js:1]: - [ ] <b>REQUIRED:</b> You overused `eslint-disable`. it's like making changes in Eslint config. they represent style guidelines that you share with your team - which is a group of all Microverse students. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/) and [Zendesk](https://microverse.zendesk.com/hc/en-us/articles/1500002085202-February-2nd-2021-linter-errors-as-a-reason-of-project-invalidation-)). kindy make sure to fix †hat in the next milestone.

**Review by rukundoeric — APPROVED** [review]
Hi @janis-jenny,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_


Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------

**Pull request merged**